exhibitions/src/main/kotlin/edu/artic/exhibitions/ExhibitionDetailFragment.kt

```diff
@@ -0,0 +1,202 @@
+                appBarLayout.expandedTitle.alpha = 0f
+                appBarLayout.expandedTitle.alpha = 1f
+            }
+
+        }
+        exhibitionImage.transitionName = exhibition.title
+    }
+
+    override fun onRegisterViewModel(viewModel: ExhibitionDetailViewModel) {
+        viewModel.setExhibitionExhibition(exhibition)
+    }
+
+    override fun setupBindings(viewModel: ExhibitionDetailViewModel) {
+        viewModel.title
+                .subscribe {
+                    expandedTitle.text = it
+                    toolbarTitle.text = it
+                }
+                .disposedBy(disposeBag)
+
+        viewModel.imageUrl
+                .subscribe {
+                    val options = RequestOptions()
+                            .dontAnimate()
+                            .dontTransform()
+                    Glide.with(this)
+//                            .asBitmap()
+                            .load(it)
+                            .apply(options)
+                            .listener(object : RequestListener<Drawable> {
+                                override fun onLoadFailed(e: GlideException?,
+                                                          model: Any?,
+                                                          target: Target<Drawable>?,
+                                                          isFirstResource: Boolean)
+                                        : Boolean {
+                                    startPostponedEnterTransition()
+                                    return false
+                                }
+
+                                override fun onResourceReady(
+                                        resource: Drawable?,
+                                        model: Any?,
+                                        target: Target<Drawable>?,
+                                        dataSource: DataSource?,
+                                        isFirstResource: Boolean): Boolean {
+                                    resource?.let {
+                                        // Adds a nice animator to scale the container down to proper aspect ratio
+                                        val parentWidth = (exhibitionImage.parent as View).width
+                                        val newHeight = if (it.intrinsicWidth > it.intrinsicHeight) {
+                                            ((it.intrinsicHeight.toFloat() / it.intrinsicWidth.toFloat()) * parentWidth).toInt()
+                                        } else {
+                                            parentWidth
+                                        }
+                                        if (newHeight != parentWidth) {
+                                            val animator = ValueAnimator.ofInt(parentWidth, newHeight)
+                                            animator.addUpdateListener {
+                                                val params = exhibitionImage.layoutParams
```

Comment:
let's kotlinize this:
```kotlin
 exhibitionImage.layoutParams = exhibitionImage.layoutParams.apply {                                         
    width = parentWidth                                               
    height = it.animatedValue as Int
  }
```
I think this'll work

Reply:
not always, sometimes it will, but unless you assign the layout params back there is a posibility that the values will not be accepted. @agrosner